refinery/core/serializers.py

```diff
@@ -34,13 +34,17 @@ def get_is_owner(self, data_set):
                 logger.error("Request is missing a user: %s", e)
                 return False
             return user_request == owner
+        else:
+            return data_set.is_owner
 
     def get_public(self, data_set):
         try:
             data_set.public
         except:
             is_public = data_set.is_public()
             return is_public
+        else:
+            return data_set.public
```

Comment:
ditto: You don't want to do too much in a `try`, but just returning isn't going cause a different exception